projects/plugins/jetpack/modules/masterbar/masterbar/class-masterbar.php

```diff
@@ -392,6 +410,79 @@ public function get_locale() {
 		return $wpcom_locale;
 	}
 
+	/**
+	 * Get Jetpack locale name.
+	 *
+	 * @param  string $slug Locale slug.
+	 * @return string Jetpack locale.
+	 */
+	public function get_jetpack_locale( $slug = '' ) {
+		if ( ! class_exists( 'GP_Locales' ) ) {
+			if ( defined( 'JETPACK__GLOTPRESS_LOCALES_PATH' ) && file_exists( JETPACK__GLOTPRESS_LOCALES_PATH ) ) {
+				require JETPACK__GLOTPRESS_LOCALES_PATH;
+			}
+		}
+
+		if ( class_exists( 'GP_Locales' ) ) {
+			$jetpack_locale_object = GP_Locales::by_field( 'slug', $slug );
+			if ( $jetpack_locale_object instanceof GP_Locale ) {
+				$jetpack_locale = $jetpack_locale_object->wp_locale;
+			}
+		}
+
+		return $jetpack_locale;
+	}
+
+	/**
+	 * Install locale if not yet available.
+	 *
+	 * @param string $locale The new locale slug.
+	 */
+	public function install_locale( $locale = '' ) {
+		if ( ! in_array( $locale, get_available_languages(), true )
+			&& ! empty( $locale ) && current_user_can( 'install_languages' ) ) {
+
+			if ( ! function_exists( 'wp_download_language_pack' ) ) {
+				require_once ABSPATH . 'wp-admin/includes/translation-install.php';
+			}
+
+			if ( ! function_exists( 'request_filesystem_credentials' ) ) {
+				require_once ABSPATH . 'wp-admin/includes/file.php';
+			}
+
+			if ( wp_can_install_language_pack() ) {
+				wp_download_language_pack( $locale );
+				load_default_textdomain( $locale );
+			}
+		}
+	}
+
+	/**
+	 * Hide language dropdown on user edit form.
+	 */
+	public function hide_language_dropdown() {
+		add_filter( 'get_available_languages', '__return_null' );
+	}
+
+	/**
+	 * Replace language dropdown with link to WordPress.com.
+	 */
+	public function replace_language_dropdown() {
+		$language_row  = printf( '<tr class="user-language-wrap"><th scope="row">' );
+		$language_row .= printf(
+			'<label for="locale">%1$s<span class="dashicons dashicons-translation" aria-hidden="true"></span></label>',
+			esc_html( __( 'Language', 'jetpack' ) )
+		);
+		$language_row .= printf( '</th><td>' );
+		$language_row .= printf(
+			'<a target="_blank" href="%1$s">%2$s</a>',
+			esc_url( 'https://wordpress.com/me/account' ),
+			esc_html( __( 'Set your profile language on WordPress.com.', 'jetpack' ) )
+		);
+		$language_row .= printf( '</td></tr>' );
+		return $language_row;
+	}
```

Comment:
FWIW, we have added a similar function in WPCMOSH for the `color-schemes`. 
https://github.com/Automattic/wpcomsh/blob/0bd23ef0b6485a0b6594df8425730015214d428c/feature-plugins/masterbar.php#L80-L89